modules/exploits/linux/http/vestacp_exec.rb

```diff
@@ -239,10 +239,13 @@ def payload_implant
 
   def exploit
     start_http_server
-    payload_implant
-    login
-    start_backup_and_trigger_payload
-    stop_service
+    begin
+      payload_implant
+      login
+      start_backup_and_trigger_payload
+    ensure
+      cleanup_service
+    end
```

Comment:
If `cleanup_service` gets called by default as part of `remote/socket_server`, I believe this isn't required?

```
>> framework.modules.create('exploits/linux/http/vestacp_exec').class.ancestors.map { |x| x.instance_method(:cleanup).source_location rescue nil }.compact
=> 
[["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/file_dropper.rb", 110],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/file_dropper.rb", 110],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/remote/http_server.rb", 228],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/remote/http_client.rb", 342],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/remote/socket_server.rb", 59],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/remote/socket_server.rb", 59],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/remote/tcp.rb", 200],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit/remote/tcp.rb", 200],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit.rb", 423],
 ["/Users/a/Documents/code/metasploit-framework/lib/msf/core/exploit.rb", 423]]
>> 

```

Reply:
Yes, I think I'd made a bad assumption about what calls what; this shouldn't be needed.